PerfNext/public/lib/js/util.js

```diff
@@ -247,6 +249,104 @@ function getEspressoPlatform() {
     return platform;
 }
 
+/* Get the OS, cputype and bits to check against the master machine list */
+function getCPUInfo() {
+    console.log('Entering getCPUInfo()');
+    
+    var platform = document.getElementById('dtextlabel').innerText;
+    console.log('platform: '+platform);
+    
+    var cpuInfo = {};
+    
+    switch (platform) {
+    
+    case 'Windows Intel x86 32 bit':
+    	cpuInfo = {cpuType: 'x86', bits: '32', OS: 'Windows'};
+    	break;
+    case 'Windows AMD x64 64 bit':
+    	cpuInfo = {cpuType: 'AMD64', bits: '64', OS: 'Windows'};
+    	break;
+    case 'AIX PPC 64 bit':
+    	cpuInfo = {cpuType: 'PPC', bits: '64', OS: 'AIX'};
+    	break;
+    case 'Linux Intel x86 32 bit':
+    	cpuInfo = {cpuType: 'x86', bits: '32', OS: 'Linux'};
+    	break;
+    case 'Linux AMD x64 64 bit':
+    	cpuInfo = {cpuType: 'AMD64', bits: '64', OS: 'Linux'};
+    	break;
+    case 'Linux PPC 32 bit':
+    	cpuInfo = {cpuType: 'PPC', bits: '32', OS: 'Linux'};
+    	break;
+    case 'Linux PPC 64 bit':
+    	cpuInfo = {cpuType: 'PPC', bits: '64', OS: 'Linux'};
+    	break;
+    case 'Linux PPCLE 64 bit':
+    	cpuInfo = {cpuType: 'PPCLE', bits: '64', OS: 'Linux'};
+    	break;
+    case 'Linux z Series 31 bit':
+    	cpuInfo = {cpuType: 'z Series', bits: '31', OS: 'Linux'};
+    	break;
+    case 'Linux z Series 64 bit':
+    	cpuInfo = {cpuType: 'z Series', bits: '64', OS: 'Linux'};
+    	break;
+    case 'MVS z Series 31 bit':
+    	cpuInfo = {cpuType: 'z Series', bits: '31', OS: 'MVS'};
+    	break;
+    case 'MVS z Series 64 bit':
+    	cpuInfo = {cpuType: 'z Series', bits: '64', OS: 'MVS'};
+    	break;
+    default:
+    	console.log('getCPUinfo() ERROR OS: '+OS);	
+    }
```

Comment:
Rather than exact string matching, is it possible to do regex here?

Reply:
Right now if we wish to use pattern matching, changes would need to be made everywhere starting from the master machine list itself and other functions still rely on that specific format. We will definitely be looking into this in the future, but for now the switch statement appears to be the easiest thing to do. An overhaul might be planned for all the functions relying on the platform drop down menu.
If you see some of the patterns: `Intel x86` is translated to just `x86` whereas `AMD x64` translates to `AMD64`  which would end up requiring cases anyway.

Reply:
I do not mean to change the format. I think it would be better to have efficient code and only deal with the spacial cases. For example,

```
const tokens = platform.split(" ");
if (token.length < 5) {
   const os = token[0];
   const bits =  token[token.length-1];
   let cpuType = token[1]+token[2];
   // special treatment for cpuType if needed:
   // remove Intel in cpuType if any
   // remove x for AMD
   // if startWith PPC, cpuType = token[1]
cpuInfo = {cpuType, bits, os };
} else {
   // error
}
```

Reply:
Agree with this, you have an opportunity to improve the code that handles these strings, changing from an inflexible 12 case statement, to the approach Lan suggests, will help future-proof this code, and still allows for special case treatment.  

Can you please try updating your implementation and verify this approach of parsing the string cases.

Reply:
Tested this implementation and it works as intended, please let me know if more changes are required. Thank you.

Reply:
Thanks @pinicman !